### ISSUE TITLE
Feature Request: Use incoming packets to render self when position reporting is off

### ISSUE THREAD

**Issue by turbo2ltr** [ge0rg/aprsdroid#96]
I am using a self contained tracker (MicroTrak from Byonics) which sends APRSDriod KISS packets that it receives as well as position packets it generates.  But if you set Location Source to manual and turn off Automatic Posting, and if your call and SSID set in APRSDroid and the MicroTrak match, your position will never get rendered on the map unless you specifically track your own call.

But if the SSID does not match I believe then it breaks notifications of messages as it does not think the messages are for you.  But I haven't thoroughly tested that yet.

In any case, It would be great to have a mode that properly handles when you generate your own position packets externally using the same SSID as what is set in the app.

**Comment by turbo2ltr** [comment]
I will test this coming weekend.  

I noticed an entry in the change log or the code (I've been trying to compile but have not had any luck.. Maybe 1.1.3?) that the behavior for position reports for ourselves was changed so they were not added to the list.  Maybe I took that out of context.

As for messaging, the messages come in without issue, but I seem to remember that notifications do not get triggered unless the message is for the configured call/ssid.  I'll test it again.

**Comment by ge0rg** [comment]
Any news on that?

**Comment by turbo2ltr** [comment]
Sorry for the delay. This has been sitting in my inbox forever and I just haven't had a chance to test.

I did confirm this behavior.  See videos for an explanation. 

This is with the latest OSM version 1.3.0-27-g7c4cd90 2015-06-02

https://www.youtube.com/watch?v=eM1sKF0G3qE

https://www.youtube.com/watch?v=Ahpt7RJJjRQ

**Comment by turbo2ltr** [comment]
A couple more tidbits..

When I'm just tracking myself (second video) I noticed my position wasn't *exactly* right, so I tested again, and it does NOT update position.  I'm not sure where that position is coming from, other than it's the first position report APRSdroid would receive.  I drove around and packets were going out, being digipeated and coming back in, but no position updates were made to the map.

My settings are set for manual position (I didn't know how else to stop APRSdriod from sending it's own updates), and that position is actually across the country in Harrisburg PA.  I looked 
on the main map to see if I was rendered there and I wasn't.

![screenshot_2015-08-29-11-08-37](https://cloud.githubusercontent.com/assets/10290656/9563686/57d3ef28-4e41-11e5-8c2c-10b9e8dd860e.png)
![screenshot_2015-08-29-11-25-45](https://cloud.githubusercontent.com/assets/10290656/9563693/78e9255c-4e41-11e5-879c-32661515ac96.png)

**Comment by ge0rg** [comment]
Thanks for the additional data. Regarding your first video, this is "by design" - just tapping the map does not enable tracking of you own position. I probably should change that, though, with some "track me" button in the map view.

Your APRSdroid configuration looks correct (Manual position and no automatic reports is the right was to use the tracker).

However, the packets you have received from your tracker are not parsed by APRSdroid correctly. Therefore it fails to update your position (and displays Aug 9th as the last update date). I've taken the raw data from aprs.fi from your callsign-ssid, and pasted it into my own APRSdroid, which it parsed correctly:

![2015-08-31_19-40-24_z1auljp6s5ej](https://cloud.githubusercontent.com/assets/165635/9585621/6b614954-5018-11e5-93dc-37f3940b189d.jpg)

This is probably an issue with the KISS transmission containing something that confuses APRSdroid. Can you please [provide a LogCat](https://github.com/ge0rg/aprsdroid/wiki/Reporting-Bugs#provide-a-logcat) of your APRSdroid receiving data from the tracker? There should be a bunch of **readPacket: ...** lines followed by **parsePacket() unsupported packet:** line followed by a stack trace. I should be able to fix the problem based on all these taken together.

**Comment by turbo2ltr** [comment]
Oh, I didn't realize it was by design.  If aprsdroid is getting position updates from the internal GPS, doesn't it render and track itself on the main map though?

Very odd.  I would think that digipeated packets would be parsed correctly since it parses all the other station's packets without issue.... but I will get you a log cat.

Seems maybe related to these changes..
https://github.com/ge0rg/aprsdroid/commit/6fab82655ac6511b13f5c9ca8ddf0a479036dd2d

**Comment by ge0rg** [comment]
Good catch! You are probably right regarding 6fab82655ac6511b13f5c9ca8ddf0a479036dd2d. I will investigate and report back soon. No need for a LogCat in the meantime.

**Comment by VK2HFG** [comment]
Has this issue been resolved?
I have what appears to be an identical issue even with the latest version 1.6.2b 2202-02-07 whereby heard position packets will only cause update of station details if the packet hasn't been digipeated, ie they have an unused wide1-1 direct from DireWolf via the KISS terminal using the R0 channel option. The issue also exists in a 2018 version that is installed on a tablet rather than the phone that I regularly use.
The updates had occured in times gone by when I used a different radio that caused tx to be echoed to rx, effectively the same as R0. I prefer that my sent pkts don't update station details, only updated from those heard after having been digipeated.
Thanks,
Graeme

**Comment by VK2HFG** [comment]
Having now found fix #62, I see the dilema. As I don't need APRSdroid to create position packets, if APRSdroid could optionally abstain from creating position packets, perhaps configured by using manual co-ordinates of zeros, and allow updates from digipeated packets if abstaining then the dilema would be overcome.  Meanwhile as my DireWolf packets contain information not possible to obtain and insert by APRSdroid yet I want displayed by APRSdroid, I'll use the Direwolf R0 option.
Graeme

**Comment by VK2HFG** [comment]
I've just found the option for automatic posting so perhaps there is already no dilema.

**Comment by VK2HFG** [comment]
No, even when APRSdroid is set to not post positions, digipeated packets of the callsign/ssid don't update the station list details.  Additionally it would be appropriate to not send a position with zero co-ordinates if the send position button is used.
